workflows/testing/test_workflow/manifest.py

```diff
@@ -0,0 +1,47 @@
+import yaml
+
+class BuildManifest:
+    @staticmethod
+    def from_file(path):
+        with open(path, 'r') as file:
+            hash = yaml.safe_load(file)
+            return BuildManifest(hash)
+
+    @staticmethod
+    def from_text(text):
+        hash = yaml.safe_load(text)
+        return BuildManifest(hash)
+
+    def __init__(self, hash):
+        print("BuildManifest")
+        print(hash)
+        schema_version = str(hash['schema-version'])
+        if schema_version != '1.0':
+            raise ValueError(f'Schema version must be 1.0; found {schema_version}')
+        self._build_id = hash['build']['build-id']
+        self._bundle_location = hash['build']['location']
+        self._components = list(map(lambda entry: Component(entry), hash['components']))
+
+    def build_id(self):
+        return self._build_id
+
+    def bundle_location(self):
+        return self._bundle_location
+
+    def components(self):
+        return self._components
+
+class Component:
```

Comment:
Save some lines with `@dataclass` or something similar?

Reply:
My python-fu is weak :) Thanks!

Reply:
Here's a similar refactor: https://github.com/opensearch-project/opensearch-build/pull/150